### DISCUSSION TITLE
No Suricata data coming sensor

### DISCUSSION THREAD

**Discussion by ihuxn4hi** [Security-Onion-Solutions/securityonion#9692] [Unsupported Versions]
I'm just getting started with Security Onion and have a sensor that I can see is logging Suricata data at /nsm/suricata/eve-*.json but there are zero logs in the manager.  Doing a tcpdump, I can see traffic going to and from the manager on port 5644 and others so I'm presuming the communication is OK.  What can I look at to troubleshoot this?

Also, is it possible to monitor more than one interface on a sensor?  I was reading https://docs.securityonion.net/en/2.3/suricata.html#configuration but didn't see anything about this.

Thanks!

**Comment by InfosecGoon** [comment]
If you have a distributed deployment, the Suricata logs are written to disk on the sensor, then picked up via Filebeat and sent to the Logstash service on the Manager, then put into a Redis Queue, and finally put into Elasticsearch. So, it could be broken anywhere after the first step. Are you running a Manager/Search node or a separate Search node?

If you want to monitor multiple physical interfaces, you can do that by using so-monitor-add to add them to the bond0 virtual interface that Security Onion is monitoring: https://docs.securityonion.net/en/2.3/so-monitor-add.html

**Reply by ihuxn4hi** [reply]
Thank you very much for your reply.

I've got separate Manager, Search, and Sensor nodes and it's running in AWS so unfortunately, there's no bond0 interface and the so-monitor-add looks like it won't work.  Is there another option for adding additional monitoring interfaces?

Given there's a separate Search node, would the sensor be shipping its logs to the Manager or the Search node?  I'm doing a tcpdump right now and not seeing any traffic to the Search node at least, although I am seeing it going to the manager on port 5644.  Although they all seem to be the same size so not sure if it's a heartbeat:

```
# tcpdump -i ens5 -c 6 -nn host 10.0.210.159 and port 5644
tcpdump: verbose output suppressed, use -v or -vv for full protocol decode
listening on ens5, link-type EN10MB (Ethernet), capture size 262144 bytes
18:53:55.651453 IP 10.0.210.159.5644 > 10.0.210.50.46848: Flags [P.], seq 1060713794:1060713829, ack 649192670, win 3858, options [nop,nop,TS val 4282191921 ecr 2982177059], length 35
18:53:55.651637 IP 10.0.210.50.46848 > 10.0.210.159.5644: Flags [.], ack 35, win 501, options [nop,nop,TS val 2982182059 ecr 4282191921], length 0
18:54:00.652365 IP 10.0.210.159.5644 > 10.0.210.50.46848: Flags [P.], seq 35:70, ack 1, win 3858, options [nop,nop,TS val 4282196921 ecr 2982182059], length 35
18:54:00.652492 IP 10.0.210.50.46848 > 10.0.210.159.5644: Flags [.], ack 70, win 501, options [nop,nop,TS val 2982187060 ecr 4282196921], length 0
18:54:05.652298 IP 10.0.210.159.5644 > 10.0.210.50.46848: Flags [P.], seq 70:105, ack 1, win 3858, options [nop,nop,TS val 4282201922 ecr 2982187060], length 35
18:54:05.652428 IP 10.0.210.50.46848 > 10.0.210.159.5644: Flags [.], ack 105, win 501, options [nop,nop,TS val 2982192060 ecr 4282201922], length 0
```

Thank you again for the help -- I really appreciate it!

**Reply by InfosecGoon** [reply]
In a distributed deployment, everything goes from the sensor to the Manager and then put into a Redis queue -- the reason for this is that the Redis queue is accessible to as many Search nodes as you want to stand up and it makes it easy to scale horizontally.

I didn't realize you were running in AWS, that adds some wrinkles. Are you using the AMI or did you do a network install? I would suggestion checking the following places for potential issues:

1. Check your AWS Security Groups between the nodes to make sure that they have access to one another on the necessary ports (listed here: https://docs.securityonion.net/en/2.3/firewall.html#node-communication)

2. Check the filebeat logs (/opt/so/log/filebeat) on the sensor to see if there are any errors connecting to the Manager.

3. Check the logstash logs (/opt/so/log/logstash) and elasticsearch logs (/opt/so/log/elasticsearch) on the Manager to see if there are any errors accepting traffic from the sensor.

4. Run so-redis-count on the Manager to see if there are documents in the queue.

5. Check the logstash and elasticsearch logs on the Search node to see if there are issues connecting to the Manager redis queue or writing to the database.

**Reply by ihuxn4hi** [reply]
Thank you so much for your response and help with this.  Your pointing me in the direction of the logs told me everything I needed to see.  On the search node, I saw this:

`[2023-02-02T22:40:15,260][WARN ][logstash.inputs.redis    ] Redis connection error {:message=>"Error connecting to Redis on so-manager:9696 (Redis::TimeoutError)", :exception=>Redis::CannotConnectError}`

I went and double checked my Security Group list of allowed inbound ports and sure enough, I missed 9696 and 9300 from the firewall list.

I see alerts coming in now so thank you very much!

Regarding having a sensor monitoring more than one interface, what would be your guidance?  I can see the interface and I can see the traffic going across it, I just don't know how to add an additional interface to be monitored.

Thank you again for your help -- I really do appreciate it.

**Reply by InfosecGoon** [reply]
What is the output from "nmcli con" on the sensor? I'd like to see your current interface configuration.

**Reply by ihuxn4hi** [reply]
Here's the output from 'nmcli con' on the sensor:

```
root@so-sensor-01 [~]
# nmcli con
NAME                UUID                                  TYPE      DEVICE  
Wired connection 1  de89b4db-e73e-3db8-88c1-a5e73837d384  ethernet  ens7    
docker0             1814d2f0-c7c7-46ab-882c-0a70296a40b7  bridge    docker0 
ens6                eadd59c6-0618-448d-aa28-d7c7afca0f2c  ethernet  ens6
``` 

Interesting that there's supposedly a difference between the ens6 and ens7 interfaces.  I've also noticed that there's an IP assigned to ens7 and none for ens6 so I'm not sure if that comes into play at all?  I'll look at those interface configurations now.

Thanks again

**Reply by InfosecGoon** [reply]
I spoke with our developer team and unfortunately AWS doesn't support interface bonding in their environment -- if you want an additional interface to accept more mirror traffic, you'll need to add another sensor node to the grid.

**Reply by ihuxn4hi** [reply]
Bummer, but thank you for letting me know.  

As a workaround, I've changed the target interface within AWS to the one that's being monitored.  I'll have to keep an eye on throughput and dropped packets.  Not ideal but better than nothing I suppose.

Thank you again for all of the help.  I really do appreciate it.

Best regards
